dwave/system/samplers/clique.py

```diff
@@ -165,17 +165,15 @@ def sample(self, bqm, chain_strength=None, **kwargs):
 
         if chain_strength is None:
             # chain length determines chain strength
-            if embedding:
-                chain_strength = 1.5 * math.sqrt(len(embedding))
+            if embedding and bqm.num_interactions > 0:
+                squared_j = (j ** 2 for j in bqm.quadratic.values())
+                rms = math.sqrt(sum(squared_j)/bqm.num_interactions)
+       
+                chain_strength = 1.5 * rms * math.sqrt(bqm.num_variables)
             else:
                 chain_strength = 1  # doesn't matter
 
-            # scale chain strength by the problem scale
-            scalar = max(max(map(abs, bqm.linear.values()), default=0),
-                         max(map(abs, bqm.quadratic.values()), default=0))
-            chain_strength *= scalar
-
-        # # scaling only make sense in Ising space
+        # scaling only make sense in Ising space
         original_bqm = bqm
 
         if bqm.vartype is not dimod.SPIN:
```

Comment:
Also, I think this should be done **before** fixing chain strength, as `bqm.quadratic.values()` are used there. If one passes a bqm with binary variables, then chain strength is not properly set

Reply:
unless `bqm.quadratic.values()` always come from a SPIN variable bqm

Reply:
You are correct, it comes after only because it didn't matter before. Excellent catch. I will fix that now